edgarpecero/OMN-Order-Service
# [OMN-30] Integrate notifications.sesClient to send emails

# [OMN-30] Integrate notifications.sesClient to send emails

---

### Summary of Changes
- Integrated Amazon SES (Simple Email Service) using `notifications.sesClient` to send email notifications directly from the application, enhancing user communication and notification capabilities.

### Task References
- [OMN-30] Integrate `notifications.sesClient` to send email notifications.

### Detailed Changes
- Added a new `notifications.sesClient` service to connect with Amazon SES for sending emails.
- Configured SES settings and sender email details to support notification functionality.
- Updated relevant components to trigger email notifications upon specific user actions (e.g., order creation or updates).
- Followed best practices for exception handling to ensure reliable email delivery and error logging.

### Testing Instructions
1. Set up SES permissions and configure the sender email in the AWS SES console.
2. Trigger an action in the application (e.g., create or update an order) that should result in an email notification.
3. Verify that the email is sent and received as expected, and check logs for any errors or failed attempts.

### Future Considerations
- Consider adding retry logic for email notifications to handle intermittent SES service issues.
- Expand email templates for additional user notifications (e.g., order shipped, account updates).
- Evaluate email volume requirements to ensure SES limits align with application demands.

---

### Additional Notes
- Ensure that SES is in production mode (not sandbox) to send emails to unverified recipients.
- Please review the email content format and configuration for any improvements.
